pkg/skel/skel.go

```diff
@@ -199,6 +200,9 @@ func validateConfig(jsonBytes []byte) *types.Error {
 	if conf.Name == "" {
 		return types.NewError(types.ErrInvalidNetworkConfig, "missing network name", "")
 	}
+	if err := utils.ValidateNetworkName(conf.Name); err != nil {
```

Comment:
If we validate networkName/containerID in **addNetwork**, then it's not necessary to validate in **skel**, because  addNetwork will call skel, this is a serial process. Validation should be done only once.

Reply:
resolved? @mccv1r0

Reply:
@mars1024 I think it's necessary to validate in both libcni and skel, because libcni is the runtime side and we want to keep errors closer to the caller and return them before we every get to execing. It's necessary in skel.go because we can't control whether the runtime itself uses libcni or not, and in fact there are alternate implementations of the CNI spec like https://github.com/containerd/go-cni